strings-csharp/StringTruncationInCSharp/StringTruncationInCSharp/StringHelper.cs

```diff
@@ -5,99 +5,111 @@ namespace StringTruncationInCSharp
-    var truncatedStringBuilder = new StringBuilder(length);
+            var length = Math.Min(maxLength, originalString.Length);
+            var truncatedStringBuilder = new StringBuilder(length);
 
-    for (int i = 0; i < length; i++)
-    {
-        truncatedStringBuilder.Append(originalString[i]);
-    }
+            for (int i = 0; i < length; i++)
+            {
+                truncatedStringBuilder.Append(originalString[i]);
+            }
 
-    var truncatedString = truncatedStringBuilder.ToString();
+            var truncatedString = truncatedStringBuilder.ToString();
+
+            return truncatedString;
+        }
 
-    return truncatedString;
-}
-        
         public string TruncateStringUsingLINQ(string originalString, int maxLength)
         {
             var truncatedString = new string(originalString.Take(maxLength).ToArray());
 
             return truncatedString;
         }
-        
-public string TruncateStringUsingForLoop(string originalString, int maxLength)
-{
-    var truncatedString = string.Empty;
-    int length = Math.Min(maxLength, originalString.Length);
 
-    for (int i = 0; i < length; i++)
-    {
-        truncatedString += originalString[i];
-    }
+        public string TruncateStringUsingForLoop(string originalString, int maxLength)
+        {
+            var truncatedString = string.Empty;
+            int length = Math.Min(maxLength, originalString.Length);
 
-    return truncatedString;
-}
-        
-public string TruncateStringUsingRegularExpressions(string originalString, int maxLength)
-{
-    if (maxLength <= 0)
-    {
-        return string.Empty;
-    }
-    else if (maxLength >= originalString.Length)
-    {
-        return originalString;
-    }
+            for (int i = 0; i < length; i++)
+            {
+                truncatedString += originalString[i];
+            }
 
-    var truncatedString = Regex.Replace(originalString, $"^(.{{0,{maxLength}}}).*$", "$1");
+            return truncatedString;
+        }
+
+        public string TruncateStringUsingRegularExpressions(string originalString, int maxLength)
+        {
+            if (maxLength <= 0)
+            {
+                return string.Empty;
+            }
+            else if (maxLength >= originalString.Length)
+            {
+                return originalString;
+            }
+
+            var truncatedString = Regex.Replace(originalString, $"^(.{{0,{maxLength}}}).*$", "$1");
+
+            return truncatedString;
+        }
+
+        public string TruncateStringUsingSpan(string originalString, int maxLength)
+        {
+            if (string.IsNullOrEmpty(originalString) || maxLength <= 0)
+            {
+                return string.Empty;
+            }
+
+            ReadOnlySpan<char> originalStringAsSpan = originalString.AsSpan();
+            if (maxLength >= originalStringAsSpan.Length)
+            {
+                return new string(originalStringAsSpan);
```

Comment:
Shouldn't we just return originalString here like in the other methods?

Reply:
Yes, I agree